### PR TITLE
Create or delete new custom licenses using license API

### PR DESCRIPTION
**Pull request by rbt-mm**
### Description
**Current Behavior:**
Today, only SPDX licenses which are already known to the system can be added to license groups or be properly resolved. Other license names uploaded by SBOM can be shown but not used in any logic. There is no possibility to create custom licenses for non-standart or own licenses.

**Proposed Behavior:**
New licenses can be created or deleted via the license API to easily specify licenses which do not have a SPDX identifier.

### Addressed Issue
#1211 

### Additional Details
- Adds a `PUT` method to the license API. The `PUT` method requires a license name and a license ID to create a new license. The newly created license will have a new persistent attribute set to true, to identify it as a custom license.
- Adds a `DELETE` method to the license API. The `DELETE` method requires the path of a license and is only able to delete licenses which are identifiable as custom licenses.
- Both methods require the `SYSTEM_CONFIGURATION` permission.

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/github-templates/CONTRIBUTING.md#pull-requests)
- [ ] ~This PR fixes a defect, and I have provided tests to verify that the fix is effective~
- [x] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)
- [ ] ~This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly~

### CODE REVIEW

File: src/main/java/org/dependencytrack/persistence/LicenseQueryManager.java [hunk 1]
```diff
@@ -152,4 +152,67 @@ License synchronizeLicense(License license, boolean commitIndex) {
...
+        license.setComment(comment);
+        license.setSeeAlso(seeAlso);
+
+        final License result = persist(license);
+
+        Event.dispatch(new IndexEvent(IndexEvent.Action.CREATE, pm.detachCopy(result)));
+        commitSearchIndex(commitIndex, License.class);
+
+        return result;
+    }
```
[the remark this answers is not in the document]
Comment (rbt-mm) [reply src/main/java/org/dependencytrack/persistence/LicenseQueryManager.java:193]: No, it is not necessarily needed. I used the methods to create projects in `ProjectQueryManager` and `ProjectResourceTest` as a guide, which used a similar method for testing, but if wanted, all tests in `LicenseResourceTest` can be rewritten to use the other existing methods and this method can be removed.

[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/persistence/LicenseQueryManager.java:193]: Yes please. Sorry if it seems pedantic, just trying to reduce the amount of code we have to maintain if possible.

[the remark this answers is not in the document]
Comment (rbt-mm) [reply src/main/java/org/dependencytrack/persistence/LicenseQueryManager.java]: There's no need to be sorry, it's totally understandable to want cleaner code 😁

**Review by nscuro — APPROVED** [review]
Thanks @rbt-mm! 🙌

**Pull request merged**